sync/category_converter.go

```diff
@@ -0,0 +1,86 @@
+package sync
+
+import (
+	"database/sql"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/appuio/appuio-cloud-reporting/pkg/db"
+	"github.com/vshn/appuio-odoo-adapter/odoo/model"
+)
+
+const (
+	elementSeparator = ":"
+)
+
+// invoiceCategoryNameRegex parses a category name in the form of "prefix Zone: zone - Namespace: namespace"
+var invoiceCategoryNameRegex = regexp.MustCompile("Zone: (\\w+) - Namespace: (\\w+)")
+
+// CategoryConverter converts category models to and from Odoo.
+type CategoryConverter struct{}
+
+// ToInvoiceCategory writes compatible fields of an existing db.Category into the given model.InvoiceCategory.
+//  The model.InvoiceCategory.ID is only set if the db.Category.Target is a non-empty string.
+//  The model.InvoiceCategory.Name field is only set if db.Category.Source is non-empty string.
+// Errors are returned if db.Category.Id is not numeric or if parsing db.Category.Source fails, however no field is set in case of errors.
+func (c CategoryConverter) ToInvoiceCategory(category *db.Category, into *model.InvoiceCategory) error {
```

Comment:
I'm not a fan of this method interface. IMO this should be 
```
ToInvoiceCategory(in db.Category) (model.InvoiceCategory, error)
```

In the current for I have no idea if the function will change anything of `category` or `into`. Also IMO a conversion function should not take a target as an input and maybe keep some fields or not. I basically need to read this function to understand what id does.